internal/engine/local/execer.go

```diff
@@ -39,3 +39,43 @@ func fakeRun(ctx context.Context, cmd model.Cmd, w io.Writer, statusCh chan Stat
 
 	statusCh <- Done
 }
+
+type processExecer struct{}
+
+func NewProcessExecer() *processExecer {
+	return &processExecer{}
+}
+
+func (e *processExecer) Start(ctx context.Context, cmd model.Cmd, w io.Writer, statusCh chan Status) chan struct{} {
+	doneCh := make(chan struct{})
+
+	go processRun(ctx, cmd, w, statusCh, doneCh)
+
+	return doneCh
+}
+
+func processRun(ctx context.Context, cmd model.Cmd, w io.Writer, statusCh chan Status, doneCh chan struct{}) {
+	defer close(doneCh)
+	defer close(statusCh)
+
+	c := exec.CommandContext(ctx, cmd.Argv[0], cmd.Argv[1:]...)
+	c.Stderr = w
+	c.Stdout = w
+
+	err := c.Start()
+	if err != nil {
+		// TODO(dmiller): should this be different status than when the command fails? Unknown?
+		statusCh <- Error
+		return
+	}
+
+	statusCh <- Running
+
+	err = c.Wait()
+	if err != nil {
+		statusCh <- Error
+		return
+	}
+
+	statusCh <- Done
```

Comment:
In my mind, I figured we'd want to `statusCh <- Error` when the process exits, whether it exited with an error or not - if a service quits with exit code 0, it's still probably not what the user wanted?

At least, that seems true for the "run a service" case. I don't know if we have other use cases where it's more like a job than a service, but I think those would be taken care of by a local resource with a cmd and no serve_cmd. But I haven't put that much thought into it and maybe I missed something.

Reply:
I think the Execer should just say it finished (and perhaps we need to includ estatus code), and then it's up to the Controller how to turn that into something that makes sense for Tilt. I've found that's a decent way to separate responsibility.

Reply:
I'll leave this as-is for now and we can come back to it later.